filesystem/squashfs/squashfs.go

```diff
@@ -379,6 +398,13 @@ func (fs *FileSystem) OpenFile(p string, flag int) (filesystem.File, error) {
 	return f, nil
 }
 
+func (fs *FileSystem) Remove(p string) error {
+	if fs.workspace == "" {
+		return filesystem.ErrNotSupported
```

Comment:
Is this the right error here? In other places, if the filesystem _supports_ the function, but not _now_ because it has become read-only, e.g. [here](https://github.com/diskfs/go-diskfs/pull/264/files#diff-c364a90e39710b015f926bf542359a918c7621bda98f597f17c53fded896918bR357-R359) we just returned:

```go
			return nil, fmt.Errorf("cannot write to read-only filesystem")
```

Granted, a specific error struct is better. 

How about create another error like you did [here](https://github.com/diskfs/go-diskfs/pull/264/files#diff-11a47696192b4602f7e0fea19870d8e2bc3be8065e259df79376c42e27a2a0bfR11-R12), something like `ErrReadOnlyFilesystem`, and then use that?

Reply:
ErrReadOnlyFilesystem is even better! Will do it that way soon.